edit user credentials

elijahdaniel opened the pull request:
Username + Email can be updated on a separate page through a PUT request to `/auth/update`

elijahdaniel committed:
updates to username successful with form

elijahdaniel committed:
styled edit button on the dashboard

elijahdaniel committed:
functionality for editing functionality and minimal styling

AceIsHuman commented on src/App.js:
```diff
@@ -1,62 +1,50 @@
-import React, { useEffect } from "react";
```
It looks like your code formatter changed a lot of things. I know we didn't discuss how our codebase should be formatted in the beginning, but large changes like these can often cause merge conflicts and make it difficult to see what was changed.

lizdoyle replied:
agree. take one more look at everything

elijahdaniel replied:
If you skim through, you can see I just added one line. No need to look through everything, but yes thank you Ace, I'll be sure to turn off the formatting next time.

AceIsHuman commented on src/components/Dashboard/Dashboard.scss:
```diff
@@ -146,3 +146,42 @@ body {
 		}
 	}
 }
+
```
You're a styling wizard! 🧙

AceIsHuman commented on src/components/Dashboard/EditAccount.js:
```diff
@@ -0,0 +1,96 @@
+		user.avatar || 'http://s3.amazonaws.com/37assets/svn/765-default-avatar.png'
+
+	const handleSubmit = e => {
+		// e.preventDefault()
```
Be sure to uncomment this code.

Keep in mind, the Redux state must be updated to display the user's new `username` or `email`.

AceIsHuman commented on src/components/Dashboard/EditAccount.js:
```diff
@@ -0,0 +1,96 @@
+		const data = {
+			username: e.target.username.value,
+			email: e.target.email.value
+		}
```
Try using `username: e.target.username.value || user.username`

This way if the input field is empty, it will use the user's current credentials instead of erasing them.

AceIsHuman commented on src/components/Dashboard/EditAccount.js:
```diff
@@ -0,0 +1,96 @@
+		axiosWithAuth()
+			.put(`/auth/update`, data)
+			.then(res => {
+				console.log(res.data)
```
Consider using `alert()` to notify the user of a successful update.

elijahdaniel committed:
update code per review

elijahdaniel commented:
- uncommented preventDefault
- added an OR `||` conditional to the data object
- added an alert as to when changes have been made to the credentials

AceIsHuman reviewed (approved):
Awesome! Ready for merging 👍

AceIsHuman committed:
Merge branch 'master' into elijah-edit-information
